### PR TITLE
feat: Add configuration options for non fatal app hangs

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

This includes configuration options for enabling iOS non-fatal app hangs and Android non-fatal ANRs.

refs: RUM-5395

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

### CODE REVIEW

File: packages/datadog_flutter_plugin/example/ios/Tests/DatadogRumPluginTests.swift [hunk 1]
```diff
@@ -165,6 +165,16 @@ class DatadogRumPluginTests: XCTestCase {
         testContracts(contracts: contracts, plugin: plugin)
     }
 
+    func testRumConfiguration_WithAppHangThreshold_IsSetCorrectly() {
+        let encoded: [String: Any?] = [
+            "applicationId": "fake-application-id",
+            "appHangThreshold": 3.2
```
Comment (plousada) [inline packages/datadog_flutter_plugin/example/ios/Tests/DatadogRumPluginTests.swift]: Could we not use a random generator here?

**Review by plousada — DISMISSED** [review]

File: packages/datadog_flutter_plugin/example/ios/Tests/DatadogRumPluginTests.swift (hunk 1 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/example/ios/Tests/DatadogRumPluginTests.swift]: 👍 Added foundation mocks.

**Review by mariusc83 — DISMISSED** [review]
LGTM

File: packages/datadog_flutter_plugin/android/src/test/kotlin/com/datadoghq/flutter/DatadogRumPluginTest.kt [hunk 2]
```diff
@@ -190,6 +192,12 @@ class DatadogRumPluginTest {
         val featureConfiguration: Any = config.getFieldValue("featureConfiguration")
         assertThat(featureConfiguration.getPrivate("sampleRate")).isEqualTo(sessionSampleRate)
         assertThat(featureConfiguration.getPrivate("trackFrustrations")).isEqualTo(trackFrustration)
+        if (trackNonFatalAnrs != null) {
+            assertThat(featureConfiguration.getPrivate("trackNonFatalAnrs")).isEqualTo(trackNonFatalAnrs)
+        } else {
+            // If null, default shouldn't be changed. Tests are run on a version that enables ANR tracking by default
+            assertThat(featureConfiguration.getPrivate("trackNonFatalAnrs")).isEqualTo(true)
```
Comment (0xnm) [inline packages/datadog_flutter_plugin/android/src/test/kotlin/com/datadoghq/flutter/DatadogRumPluginTest.kt:199]: Minor API suggestion

```suggestion
            assertThat(featureConfiguration.getPrivate("trackNonFatalAnrs")).isFalse
```

**Review by 0xnm — DISMISSED** [review]

**Review by 0xnm — DISMISSED** [review]

File: packages/datadog_flutter_plugin/android/src/test/kotlin/com/datadoghq/flutter/DatadogRumPluginTest.kt (hunk 2 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/android/src/test/kotlin/com/datadoghq/flutter/DatadogRumPluginTest.kt:199]: Actually it turns out I don't have `isFalse`?

Comment (0xnm) [reply packages/datadog_flutter_plugin/android/src/test/kotlin/com/datadoghq/flutter/DatadogRumPluginTest.kt:199]: ah, maybe `isFalse()` then?

Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/android/src/test/kotlin/com/datadoghq/flutter/DatadogRumPluginTest.kt:199]: Nope (I checked). I don't know why but my library doesn't have anything like `isFalse`

**Review by 0xnm — DISMISSED** [review]

**Review by 0xnm — APPROVED** [review]

**Pull request merged**